src/algorithm.jl

```diff
@@ -22,14 +22,17 @@ Run the algorithm `algo` on the stochastic program `sp` until the termination cr
 function optimize!(sp::AbstractStochasticProgram, algo::AbstractAlgorithm, stopcrit::AbstractStoppingCriterion=IterLimit(), verbose=0)
     # Default implementation, define a specific method for algorithms for which
     # this default is not appropriate
-    paths = nothing
     info = Info()
-    while (paths === nothing || getstatus(paths[1]) != :Infeasible) && !stop(stopcrit, info)
-        @timeit info.timer "iteration $(niterations(info)+1)" paths, result = iterate!(sp, algo, info.timer, verbose)
+    while !stop(stopcrit, info)
+        @timeit info.timer "iteration $(niterations(info)+1)" result = iterate!(sp, algo, info.timer, verbose)
+        paths = result.paths
         push!(info.results, result)
         if verbose >= 3
             print_iteration_summary(info)
         end
+        if getstatus(paths[1]) != :Infeasible
```

Comment:
You can do `getstatus(result.paths[1].sol_scenario[1])` here directly and not create the `paths` variable.